src/deface.py

```diff
@@ -110,20 +188,26 @@ def run_afni_refacer(primary_t1, others, subj_input_dir, sess_dir, output_dir):
         return missing_refacer_out
 
 
-def deface_primary_scan(subj_input_dir, sess_dir, mapping_dict, output_dir):
+def deface_primary_scan(input_bids_dir, subj_input_dir, sess_dir, mapping_dict, output_dir, no_clean):
     missing_refacer_outputs = []  # list to capture missing afni refacer workdirs
 
-    subj_id = subj_input_dir.name
-    sess_id = sess_dir.name if sess_dir else None
+    subj_id = os.path.basename(subj_input_dir)
+    sess_id = os.path.basename(sess_dir) if sess_dir else None
```

Comment:
This is a minor comment/question. Is there a reason you're using os.path.basename() instead of the pathlib function which does the same thing?

Reply:
I forgot the correct command and didn't want to look it up, so I used what I knew, which you're right does the same job. Also, if I remember right the subj_input_dir and sess_dir variables are strings coming into those arguments and not Path objects.